airflow/models.py

```diff
@@ -1996,7 +1996,14 @@ def execute(self, context):
 
         Refer to get_template_context for more context.
         """
-        raise NotImplementedError()
+        if self.dag:
+            os.environ['AIRFLOW_DAG_ID'] = self.dag.dag_id
+        dagrun = context['dag_run']
+        if dagrun and dagrun.execution_date:
+                os.environ['AIRFLOW_DAGRUN'] = dagrun.execution_date.isoformat()
```

Comment:
Not sure what happened to my old comments on github, but here they are again:
I think we should add an _EXECUTION_DATE suffix to AIRFLOW_DAGRUN (and same for TASK_INSTANCE below) in case we ever want other things from DAGRUN as env variables e.g. _START_DATE.